bpm/src/test/java/org/jboss/pnc/bpm/test/RepositoryManagerResultSerializationTest.java

```diff
@@ -16,41 +16,91 @@
+import org.junit.Before;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.junit.MockitoJUnitRunner;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.io.IOException;
+import java.lang.reflect.Field;
+
+import static org.mockito.Mockito.when;
 
 /**
  * Test serialization of Repository manager rest
  */
+@RunWith(MockitoJUnitRunner.class)
 public class RepositoryManagerResultSerializationTest {
 
     private final Logger log = LoggerFactory.getLogger(RepositoryManagerResultSerializationTest.class);
 
+
+    @Mock
+    Configuration configuration;
+
+    @Spy
+    TargetRepositoryMapper targetRepositoryMapper;
+
+    @Spy
+    AbstractArtifactMapperImpl artifactMapper;
+
+    @Spy
+    @InjectMocks
+    RepositoryManagerResultMapper repositoryManagerResultMapper;
+
+    @Before
+    public void before() throws Exception{
+        IndyRepoDriverModuleConfig indyRepoDriverModuleConfig = new IndyRepoDriverModuleConfig("http://url.com");
+        indyRepoDriverModuleConfig.setExternalRepositoryMvnPath("http://url.com");
+        indyRepoDriverModuleConfig.setExternalRepositoryNpmPath("http://url.com");
+        indyRepoDriverModuleConfig.setInternalRepositoryMvnPath("http://url.com");
+        indyRepoDriverModuleConfig.setInternalRepositoryNpmPath("http://url.com");
+        when(configuration.getModuleConfig(new PncConfigProvider<>(IndyRepoDriverModuleConfig.class))).thenReturn(indyRepoDriverModuleConfig);
+        injectMethod("config", artifactMapper, configuration, AbstractArtifactMapper.class);
+        injectMethod("targetRepositoryMapper", artifactMapper, targetRepositoryMapper, AbstractArtifactMapperImpl.class);
+    }
+
+    private void injectMethod(String fieldName, Object to, Object what, Class clazz) throws NoSuchFieldException, IllegalAccessException {
+        Field f = clazz.getDeclaredField(fieldName);
+        f.setAccessible(true);
+        f.set(to, what);
+    }
+
     @Test
-    public void serializeAndDeserializeRepositoryManagerResult() throws IOException, BuildDriverException {
+    public void serializeAndDeserializeRepositoryManagerResult() throws IOException {
 
         ObjectMapper mapper = new ObjectMapper();
+        mapper.registerModule(new JavaTimeModule());
 
         RepositoryManagerResult repoMngrResult = RepositoryManagerResultMock.mockResult();
-        RepositoryManagerResultRest repoMngrResultRest = new RepositoryManagerResultRest(repoMngrResult);
+        RepositoryManagerResultRest repoMngrResultRest = repositoryManagerResultMapper.toDTO(repoMngrResult);
 
         String repoMngrResultJson = JsonOutputConverterMapper.apply(repoMngrResultRest);
         log.debug("RepoManagerResultJson : {}", repoMngrResultJson);
 
         RepositoryManagerResultRest deserializedRepoManResultRest = mapper.readValue(repoMngrResultJson, RepositoryManagerResultRest.class);
```

Comment:
Maybe replace with JsonOutputConverterMapper?

Reply:
Could you elaborate? 
Do you mean to use:
`JsonOutputConverterMapper.apply(repositoryManagerResultMapper.toDTO(repoMngrResult))`

Reply:
`JsonOutputConverterMapper.readValue(repoMngrResultJson, RepositoryManagerResultRest.class);`

Reply:
Oh you mean to delete ObjectMapper!